app/src/main/java/ru/myuniquenickname/myapplication/RecyclerViewAdapterMovies.kt

```diff
@@ -22,23 +27,27 @@ class RecyclerViewAdapterMovies(private val clickListener: OnRecyclerItemClicked
     }
 
     override fun onBindViewHolder(holder: RecyclerViewViewHolder, position: Int) {
-        val itemMovies = DataSource.listMovies[position]
-        holder.binding.apply {
-            maskFragmentList.setImageResource(itemMovies.imageResourceMaskList)
-            maskFragmentList.setImageResource(itemMovies.imageResourceMaskList)
-            logoFragmentList.setImageResource(itemMovies.imageResourceLogoList)
-            ageFragmentList.text = itemMovies.age.toString() + "+"
-            name.text = itemMovies.name
-            tagFragmentList.text = itemMovies.tag
-            reviewsFragmentList.text = itemMovies.reviews.toString() + " REVIEWS"
-            durationMin.text = itemMovies.minutes.toString() + " MIN"
-            if (itemMovies.like)
-                imageViewLike.imageTintMode = PorterDuff.Mode.SRC_IN
-            else
-                imageViewLike.imageTintMode = PorterDuff.Mode.DST_IN
-            imageViewLike.setOnClickListener { onClickLike(imageViewLike, position) }
+        if (listMovies != null) {
+            val itemMovies = listMovies[position]
+            holder.binding.apply {
```

Comment:
Rewrite it cleaner. You might put the poster and like settings to separate functions and call them here.